core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -3527,6 +3527,38 @@ private void checkPushJoinThroughUnionOnRightDoesNotMatchSemiOrAntiJoin(JoinRelT
         .check();
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-6880">
+   * [CALCITE-6880] Implement IntersectToSemiJoinRule</a>. */
```

Comment:
Please add a test case where the intersected sets do not have exactly the same column types. Coercion should take care of it, but let's make sure.

Reply:
> Please add a test case where the intersected sets do not have exactly the same column types. Coercion should take care of it, but let's make sure.

OK, I will add test for it.